site/src/components/layout/rectangles-example-measure.js

```diff
@@ -112,35 +112,20 @@ var data = [
 ];
 
 //START
-// measure the dimension of the given text
-function boundingBox(textToMeasure) {
-    var svg = d3.select('body').append('svg')
-        .attr('width', 1000)
-        .attr('height', 1000);
-    var text = svg.append('text')
-        .text(textToMeasure);
-    var bbox = text[0][0].getBBox();
-    svg.remove();
-    return [bbox.width, bbox.height];
-}
+var labelMargin = 2;
 
-// a very simple example component
-function label(selection) {
-    selection.append('rect')
-            .layout('flex', 1);
-    selection.append('text')
-            .text(function(d) { return d.language; })
-            .layout({
-                position: 'absolute',
-                bottom: 0
-            })
-            .attr({
-                // vertically centre the text in the container
-                'dominant-baseline': 'middle',
-                'dy': '-0.5em'
-            });
-    selection.layout();
-}
+var label = fc.tool.container()
+    .padding(labelMargin)
+    .component(function(sel) {
```

Comment:
You're missing one small step ...


```
.decorate(function(g) {
  g.enter()
    .select('g')
    .append('text')
    .attr('dy', '0.7em')
    .text(function(d) { return d.language; });
})
```

You need to select the inner `g` element that has the padding. It's a small thing, but couples this code to the internal structure.